SimplifiedTests/NYPLCatalogUngroupedFeedTests.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  NYPLCatalogUngroupedFeedTests.swift
+//  Simplified
+//
+//  Created by Ernest Fan on 2022-01-04.
+//  Copyright © 2022 NYPL Labs. All rights reserved.
+//
+
+import XCTest
+@testable import SimplyE
+
+class NYPLCatalogUngroupedFeedTests: XCTestCase {
+  
+  var feedProvider: NYPLOPDSFeedMock.Type!
+  
+  override func setUpWithError() throws {
+    try super.setUpWithError()
+    feedProvider = NYPLOPDSFeedMock.self
+    feedProvider.numberOfFetchAllowed = 15
+    feedProvider.shouldReturnFeedWithSupportedBooks = true
+    feedProvider.shouldReturnFeed = true
+  }
+
+  override func tearDownWithError() throws {
+    try super.tearDownWithError()
+    feedProvider = nil
+  }
+
+  func testCatalogFeedWithSupportedBooks() throws {
+    let testFeedUrl = Bundle.init(for: NYPLCatalogUngroupedFeedTests.self)
+      .url(forResource: "NYPLCatalogUngroupedFeedWithSupportedBooks", withExtension: "xml")!
+    
+    let expectation = self.expectation(description: "fetching")
+    var result: NYPLCatalogUngroupedFeed?
+    
+    NYPLCatalogUngroupedFeed.fetchSupportedBooks(with: testFeedUrl,
+                                                 provider: feedProvider) { feed in
+      result = feed
+      expectation.fulfill()
+    }
+    
+    waitForExpectations(timeout: 3, handler: nil)
+    
+    guard let result = result else {
+      XCTFail()
+      return
+    }
+    
+    XCTAssertEqual(result.books.count, 10)
+  }
+
+  func testCatalogFeedWithUnsupportedBooksUntilNoNextURLAvailable() throws {
+    let testFeedUrl = Bundle.init(for: NYPLCatalogUngroupedFeedTests.self)
+      .url(forResource: "NYPLCatalogUngroupedFeedWithUnsupportedBooks", withExtension: "xml")!
+    
+    let expectation = self.expectation(description: "fetching")
+    var result: NYPLCatalogUngroupedFeed?
+    
+    feedProvider.shouldReturnFeedWithSupportedBooks = false
+    feedProvider.numberOfFetchAllowed = 2
+    
+    NYPLCatalogUngroupedFeed.fetchSupportedBooks(with: testFeedUrl,
+                                                 provider: feedProvider) { feed in
+      result = feed
+      expectation.fulfill()
+    }
+    
+    waitForExpectations(timeout: 10, handler: nil)
```

Comment:
do we need a timeout this long, and if so, why? usually 3 seconds or so are enough.